File: lib/model/src/main/kotlin/dk/example/feedback/model/enumerations/CalendarProvider.kt
package dk.example.feedback.model.enumerations

enum class CalendarProvider {
    GOOGLE,
    APPLE,
    MICROSOFT,
    ZOOM;

    companion object {
        fun fromProdId(prodId: String?): CalendarProvider? {
            val value = prodId?.lowercase() ?: return null
            return when {
                "google" in value -> GOOGLE
                "apple" in value -> APPLE
                "microsoft" in value -> MICROSOFT
                "zoom" in value -> ZOOM
                else -> null
            }
        }
    }
}

File: ios-app/Modules/Sources/Domain/Models/CalendarProvider.swift
import Foundation
import Utility

public enum CalendarProvider: String, Codable, Equatable, RawRepresentable, Sendable, CaseIterable {
    case GOOGLE, APPLE, MICROSOFT, ZOOM
}

extension CalendarProvider {
    public init(_ input: String) {
        guard let feedbackType = CalendarProvider(rawValue: input) else {
            fatalError("Could not parse \(input) into a valid FeedbackType")
        }
        self = feedbackType
    }
}
